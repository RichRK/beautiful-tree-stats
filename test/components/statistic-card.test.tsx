import React from 'react'
import { screen, render } from '@testing-library/react'
import StatisticCard from '../../components/statistic-card'

const amount = 11
const emoji = '🏞'
const title = 'National parks in France'

const component = <StatisticCard amount={amount} emoji={emoji} title={title} />

describe('StatisticCard component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(component, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders amount and title', () => {
    render(component)
    screen.getByText(title)
    screen.getByText(amount)
  })

  it('emoji is not italicised', () => {
    render(component)
    expect(screen.getByText(emoji).classList.contains('not-italic')).toBe(true)
  })
})
