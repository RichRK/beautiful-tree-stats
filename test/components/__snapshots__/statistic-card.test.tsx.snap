// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center text-center shadow-md border-1 border-gray-400 p-2 sm:p-6 md:p-8 rounded-md"
  >
    <p
      class="text-4xl md:text-5xl font-bold mb-2 md:mb-3"
    >
      11
    </p>
    <div
      class="flex justify-center"
    >
      <h3
        class="italic"
      >
        National parks in France
        <span
          class="ml-1.5 not-italic"
        >
          🏞
        </span>
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
